### ISSUE TITLE
Failure to copy ownership permissions to remote folder mounted with sshfs

### ISSUE THREAD

**Issue by Kabouik** [jarun/nnn#458]
#### Environment details 

[x] Operating System: Solus 4
[x] Desktop Environment: Budgie Desktop
[x] Terminal Emulator: Tilix and Gnome terminal
[x] Shell: Bash
[ ] Custom desktop opener (if applicable):
[x] Issue exists on `nnn` master: yes

#### Exact steps to reproduce the issue
1. Mount a remote folder using `c`, then `s` for sshfs and type relevant user1@ip and password.
2. Copy local files with permissions `664` to a remote folder owned by `user1` (`nemo` in the example below).

```
nemo@192.168.1.29's password: 
cp: failed to preserve ownership for "./Donovan/The Hurdy Gurdy Man/1.01. Donovan - Hurdy Gurdy Man.mp3": Permission denied
```
The file is copied, but its permissions go from `664` in the local folder to `600` in the remote folder (and parent folders copied with the file go from `775` to `700`). Here are the file properties after it has been pasted in the remote folder:

```
  File: /home/mathieu/.config/nnn/nemo@192.168.1.29/Music/Donovan/The Hurdy Gurdy Man/1.01. Donovan - Hurdy Gurdy Man.mp3
  Size: 7847696         Blocks: 15328      IO Block: 4096   regular file
Device: 2dh/45d Inode: 102         Links: 1
Access: (0600/-rw-------)  Uid: (100000/ UNKNOWN)   Gid: (100000/ UNKNOWN)
Access: 2020-01-31 22:24:23.000000000 +0100
Modify: 2020-01-31 22:24:20.000000000 +0100
Change: 2020-01-31 22:24:20.000000000 +0100
 Birth: -
```

*versus* the original file in the local folder:

```
  File: /media/stockage/Media/MUSIC/Donovan/The Hurdy Gurdy Man/1.01. Donovan - Hurdy Gurdy Man.mp3
  Size: 7847696         Blocks: 15328      IO Block: 4096   regular file
Device: 801h/2049d      Inode: 3933962     Links: 1
Access: (0664/-rw-rw-r--)  Uid: ( 1000/ mathieu)   Gid: ( 1000/ mathieu)
Access: 2020-01-31 22:24:23.744808304 +0100
Modify: 2020-01-31 22:24:20.730805500 +0100
Change: 2020-01-31 22:24:20.730805500 +0100
 Birth: -
```

**Comment by jarun** [comment]
It depends on the SSHFS options you use to mount the remote. You can use custom options with `nnn` or set them in `~/.ssh/config`. `nnn` does have the `--preserve` option of `cp` set when it copies files.

Here's an example: https://superuser.com/questions/964821/translate-permissions-with-as-sshfs-remote-mount

**Comment by Kabouik** [comment]
Oh, thanks. I don't have much experience with sshfs actually. I need to find a way to set `allow_other,default_permissions` in `~/.ssh/config` now, I'm not sure how to do that yet since it seems the configuration file has no options for these parameters. The comment below is also a bit concerning:
> Be sure that usernames are the same across machines though or you might allow someone to write into the home directory of another machine of an account he or she doesn't own.
I was assuming everything written on the remote folder would be owned by the username used when connecting to the ip.

**Comment by 0xACE** [comment]
I have this in my `.bashrc` adjust it to your needs.

`export NNN_SSHFS_OPTS="sshfs -o follow_symlinks"`

ELI5: `sshfs` is essentially mapping `scp` commands to your file structure. If the `user` you are connecting as has access to a file, you will be given the same access.

I'm not 100% sure about the quote you added, you'd have to read further on `sshd` and co to understand what it means.

**Comment by jarun** [comment]
> export NNN_SSHFS_OPTS="sshfs -o follow_symlinks"

Just a note. If you are on master, the env var would be: `NNN_SSHFS`

**Comment by 0xACE** [comment]
Haha, a bit off topic, but seems like this code:
```
		char *ace_cmd = "sshfs -o follow_symlinks";
		flag |= F_MULTI;
		env = ace_cmd;
```
Is the reason I didn't detect that change

**Comment by jarun** [comment]
Clean up some of your patches. ;)

**Comment by Kabouik** [comment]
Thanks, I will try `export NNN_SSHFS="sshfs -o follow_symlinks,allow_other,default_permission" instead of the `~/.ssh/config` way.

**Comment by 0xACE** [comment]
I'm not sure if they accept `-o follow_symlink,...` maybe they have to be seperated completely `-o follow -o allow_other -o default_permission` etc... you get the point

**Comment by 0xACE** [comment]
#460 There you go bud. Now i'm really heading to bed.
